### PR TITLE
Improve TileSetAtlasSourceEditor UI

### PR DESCRIPTION
**Pull request by timothyqiu**
Based on my mockup posted on RocketChat: https://chat.godotengine.org/channel/editor?msg=DAb8wAnQ4z3LkyDnE

* The three mode buttons are moved to the top and texts are added. This makes it easier to know what you can do / what you are doing.
* Uses inspector category instead of a Label to show what's being edited. This also allows better scrolling behavior.

<table>
<tr><th>Before</th><td>

before
</td></tr>
<tr><th>After</th><td>

after
</td></tr>
</table>

---

The subtle vertical movement of main area when one tile is selected/deselected is caused by the top right tile info label showing / hiding. It's an issue that existed before this PR. I tried to fix it in this PR, but there does not seem to be a nice approach.

### CODE REVIEW

**Comment by groud** [comment]
I think the three buttons should be part of the left column so that (maybe?) the separator can be removed, and to avoid the weird gap between the separator and the Container's split in the bottom.

Aside from that, I wasn't sure that using the inspector categories was a good idea. Initially I used different inspector to edit different object in the same place (like, the tile info and the atlas one, something like that). But now it seems that, in the end, we don't have situations where two objects are edited at the same time in the same column, so using a category as title is fine.

**Comment by timothyqiu** [comment]
Made the three buttons part of the left column.

But I still think it'll cause issue reports about that column being too wide in some languages.

File: editor/plugins/tiles/tile_set_atlas_source_editor.cpp [hunk 1]
```diff
@@ -97,6 +105,8 @@ bool TileSetAtlasSourceEditor::TileSetAtlasSourceProxyObject::_get(const StringN
 }
 
 void TileSetAtlasSourceEditor::TileSetAtlasSourceProxyObject::_get_property_list(List<PropertyInfo> *p_list) const {
+	p_list->push_back(PropertyInfo(Variant::NIL, TTR("Atlas"), PROPERTY_HINT_NONE, String(), PROPERTY_USAGE_CATEGORY));
+	p_list->push_back(PropertyInfo(Variant::INT, "id", PROPERTY_HINT_RANGE, "0," + itos(INT_MAX) + ",1"));
```
Comment (groud) [inline editor/plugins/tiles/tile_set_atlas_source_editor.cpp:109]: There's not reason to move the ID property here I believe, no ? Like, you could simply use `ADD_PROPERTY(PropertyInfo(Variant::NIL, TTR("Atlas"), PROPERTY_HINT_NONE, String(), PROPERTY_USAGE_CATEGORY))` just before `ADD_PROPERTY(PropertyInfo(Variant::INT, "id", PROPERTY_HINT_RANGE, "0," + itos(INT_MAX) + ",1"), "set_id", "get_id");` instead.

That would avoid some unnecessary changes IMO.

File: editor/plugins/tiles/tile_set_atlas_source_editor.cpp [hunk 2]
```diff
@@ -381,12 +385,13 @@ void TileSetAtlasSourceEditor::AtlasTileProxyObject::_get_property_list(List<Pro
 	}
 
 	// ID and size related properties.
+	p_list->push_back(PropertyInfo(Variant::NIL, TTR("Tile"), PROPERTY_HINT_NONE, String(), PROPERTY_USAGE_CATEGORY));
```
Comment (groud) [inline editor/plugins/tiles/tile_set_atlas_source_editor.cpp:388]: Maybe it should be changed to Alternative Tile when an alternative tile is selected.

File: editor/plugins/tiles/tile_set_atlas_source_editor.cpp [hunk 3]
```diff
@@ -2578,17 +2563,17 @@ TileSetAtlasSourceEditor::TileSetAtlasSourceEditor() {
 	tile_atlas_view->add_control_over_alternative_tiles(alternative_tiles_control_unscaled, false);
 	alternative_tiles_control_unscaled->set_mouse_filter(Control::MOUSE_FILTER_IGNORE);
 
-	tile_atlas_view_missing_source_label = memnew(Label);
```
Comment (groud) [inline editor/plugins/tiles/tile_set_atlas_source_editor.cpp:2581]: If I understand correctly, this was unused right ? If so removing it is fine.

**Review by groud — CHANGES_REQUESTED** [review]

File: editor/plugins/tiles/tile_set_atlas_source_editor.cpp (hunk 1 above)
Comment (timothyqiu) [reply editor/plugins/tiles/tile_set_atlas_source_editor.cpp:109]: Properties added with `ADD_PROPERTY` comes before ones defined here. I moved ID here so it's after the category entry.

File: editor/plugins/tiles/tile_set_atlas_source_editor.cpp (hunk 3 above)
Comment (timothyqiu) [reply editor/plugins/tiles/tile_set_atlas_source_editor.cpp:2581]: Yes, it's never used.

File: editor/plugins/tiles/tile_set_atlas_source_editor.cpp (hunk 2 above)
Comment (timothyqiu) [reply editor/plugins/tiles/tile_set_atlas_source_editor.cpp]: I changed this so it's "Base Tile" or "Alternative Tile" when only one tile is selected, and it's "Tiles" when multiple tiles are selected.

File: editor/plugins/tiles/tile_set_atlas_source_editor.cpp (hunk 1 above)
Comment (groud) [reply editor/plugins/tiles/tile_set_atlas_source_editor.cpp:109]: Well, yeah, but the category can also be defined with `ADD_PROPERTY`. So there's no need to move it here, you can simply move category creation to the bind_method function.

Comment (timothyqiu) [reply editor/plugins/tiles/tile_set_atlas_source_editor.cpp:109]: `ADD_PROPERTY` needs a getter and setter, so a category can't be created with it. ClassDB might allow using empty string as getter/setter, but it feels hacky and there are no such usage in the codebase.

Comment (groud) [reply editor/plugins/tiles/tile_set_atlas_source_editor.cpp:109]: Ah ok, yeah that explains it then. Ok!

**Review by groud — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!
